### PR TITLE
Improve Frontend development infra

### PR DESCRIPTION
**Pull request by pbadillatorrealba**
The goal of this branch was to improve the frontend development infrastructure. Three major changes were made:

- The package manager was changed to yarn, which is generally recognized as a better option (higher speed, stability and security than npm). This can have the unexpected advantage that the build generated on the front end is faster than when using npm.
- Modified eslint to be less restrictive.
- Set prettier as formatter, to standardize styles.

In addition, the development documentation was improved by providing more detailed guidelines, for both front and back end, on how to prepare the development environments, how to run and how to use the linting, testing and formatting tools.

Unfortunately, when implementing prettier, several styles of various js, jsx and jsx files had to be changed. However, there were no implementation changes.

Changes:

- Added frontend development documentation.
- improved backend development documentation.
- Change front package management to yarn.
- Improved gitignore for frontend.
- Added missing packages that allow the front end to run locally
- Added prettier as default frontend formatter for the project.
- Formatted all front end files according to prettier styles.
- Updated front build action to use yarn instead npm.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -44,53 +45,129 @@ Finally, go to `http://localhost:3000/ <http://localhost:3000/>` in your browser
...
+    $ yarn start
+
+
+Linting and formatting
+~~~~~~~~~~~~~~~~~~~~~~
+
+The project uses as default linter `eslint <https://eslint.org/>`_ with 
+the `react/recommended`, `standard-with-typescript`` and `prettier`` styles. 
+
+To run manually the linter, execute:
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
To manually run the linter, execute on `DashAI/front`:
```

File: README.rst [hunk 2]
```diff
@@ -99,13 +176,42 @@ or
...
+    without making changes to the main database.
+
+
+Linting and formatting
+~~~~~~~~~~~~~~~~~~~~~~
+
+The project uses as default backend linter 
+`ruff <https://github.com/charliermarsh/ruff>`_:
+
+To run manually the linter, execute
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
To manually run the linter, execute
```

File: README.rst [hunk 3]
```diff
@@ -99,13 +176,42 @@ or
...
+
+.. code:: bash
+
+    $ ruff .
+
+
+The project uses `prettier <https://prettier.io/>`_ as default formatter. 
+
+To format the code manually, execute:
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
To manually format the code, execute:
```

File: README.rst [hunk 4]
```diff
@@ -99,13 +176,42 @@ or
...
+`ruff <https://github.com/charliermarsh/ruff>`_:
+
+To run manually the linter, execute
+
+.. code:: bash
+
+    $ ruff .
+
+
+The project uses `prettier <https://prettier.io/>`_ as default formatter.
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
The project uses `black <https://black.readthedocs.io/en/stable/>`_ as default formatter. 
```

File: README.rst [hunk 5]
```diff
@@ -99,13 +176,42 @@ or
...
+    
+    The database session is parametrized in every endpoint as 
+    ``db: Session = Depends(get_db)`` so we can test endpoints on a test database 
+    without making changes to the main database.
+
+
+Linting and formatting
+~~~~~~~~~~~~~~~~~~~~~~
+
+The project uses as default backend linter
```
Comment (cristian-tamblay) [inline README.rst:215]: Where do we run the following code? `DashAI/back` or on root

File: README.rst [hunk 6]
```diff
@@ -99,13 +176,42 @@ or
...
-Testing
--------
 
-To test the software, run from the root directory:
+Execute tests
+~~~~~~~~~~~~~
+
+DashAI uses `pytest <https://docs.pytest.org/>`_ to perform the backend
+tests.
+To execute the backend tests run:
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
To execute the backend tests run from the root directory:
```

File: README.rst [hunk 7]
```diff
@@ -44,53 +45,129 @@ Finally, go to `http://localhost:3000/ <http://localhost:3000/>` in your browser
...
 
-    $ pip install -r requirements.txt
-    $ pip install -r requirements-dev.txt
+    $ yarn install
+
+
+Running the frontend
+~~~~~~~~~~~~~~~~~~~~~~
+
+Launch the front-end development server by running the following command:
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
Launch the front-end development server by running the following command on `DashAI/front`:
```

File: README.rst [hunk 8]
```diff
@@ -44,53 +45,129 @@ Finally, go to `http://localhost:3000/ <http://localhost:3000/>` in your browser
...
+
+Prepare the environment
+~~~~~~~~~~~~~~~~~~~~~~~
+
+1. `Install the LTS node version <https://nodejs.org/en>`_. 
+
+2. Install `Yarn` package manager following the instructions located on the
+   `yarn getting started <https://yarnpkg.com/getting-started>`_ page.
+
+3. Install the project packages using yarn:
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
3. Install the project packages using yarn on `DashAI/front`: 
```

File: README.rst [hunk 9]
```diff
@@ -44,53 +45,129 @@ Finally, go to `http://localhost:3000/ <http://localhost:3000/>` in your browser
...
+
 
-Running DashAI
---------------
+Running the Backend
+~~~~~~~~~~~~~~~~~~~
 
 There are two ways to run DashAI:
 
+1. By executing DashAI as a module:
```
Comment (cristian-tamblay) [inline README.rst:170]:
```suggestion
1. By executing DashAI as a module from the root directory:
```

File: README.rst [hunk 10]
```diff
@@ -44,53 +45,129 @@ Finally, go to `http://localhost:3000/ <http://localhost:3000/>` in your browser
...
+    
+    $ yarn prettier --write src
 
 
 Build the frontend
-------------------
+~~~~~~~~~~~~~~~~~~
 
-Install node and npm. Then, go to DashAI/front:
+Execute:
```
Comment (cristian-tamblay) [inline README.rst]:
```suggestion
Execute from `DashAI/front`:
```

File: DashAI/front/.eslintrc.js [hunk 11]
```diff
@@ -1,23 +1,22 @@
...
+    ],
+    "parserOptions": {
+        "ecmaVersion": "latest",
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react"
+    ],
+    "rules": {
+    }
```
Comment (cristian-tamblay) [inline DashAI/front/.eslintrc.js]: We need to add this rule for our windows devs. We can't introduce this problem again
```suggestion
    rules: {
    'linebreak-style': ['error', process.platform === 'win32' ? 'windows' : 'unix']
  },
```

File: DashAI/front/.editorconfig [hunk 12]
```diff
@@ -0,0 +1,10 @@
+root = true
+
+[*]
+end_of_line = lf
```
Comment (cristian-tamblay) [inline DashAI/front/.editorconfig:4]: I hope this lets windows users use CRLF line end when building

File: DashAI/front/package.json [hunk 13]
```diff
@@ -2,18 +2,20 @@
   "name": "dashai-frontend",
   "version": "0.1.0",
   "private": true,
+  "packageManager": "yarn@3.5.0",
   "dependencies": {
-    "@mui/material": "^5.11.13",
+    "@emotion/react": "^11.10.6",
+    "@emotion/styled": "^11.10.6",
```
Comment (cristian-tamblay) [inline DashAI/front/package.json:8]: Are we adding emotion? I remember @maximi4321 talked about using styled-components. Please confirm it before adding another CSS-in-JS library

File: README.rst [hunk 14]
```diff
@@ -44,53 +45,129 @@ Finally, go to `http://localhost:3000/ <http://localhost:3000/>` in your browser
...
+2. Install `Yarn` package manager following the instructions located on the
+   `yarn getting started <https://yarnpkg.com/getting-started>`_ page.
+
+3. Install the project packages using yarn: 
 
 .. code:: bash
 
-    $ pip install -r requirements.txt
-    $ pip install -r requirements-dev.txt
+    $ yarn install
```
Comment (cristian-tamblay) [inline README.rst:85]:
```suggestion
    $ cd DashAI/front && yarn install
```

**Review by cristian-tamblay — CHANGES_REQUESTED** [review]
Some typos/clarity on README.
If we switch to yarn, we need to make sure it works flawlessly on windows, we already fixed the problem of npm run build and CLRF on windows.

File: DashAI/front/package.json (hunk 13 above)
Comment (pbadillatorrealba) [reply DashAI/front/package.json:8]: As discussed, we will gradually replace styled components with emotion.

File: README.rst (hunk 5 above)
Comment (pbadillatorrealba) [reply README.rst:215]: I removed the warnings and added a step to each instruction to indicate whether or not to move to each path.
Maybe later we should think about making a makefile.

File: DashAI/front/.editorconfig (hunk 12 above)
Comment (pbadillatorrealba) [reply DashAI/front/.editorconfig:4]: This file configures the text editor to set line breaks to lf by default. But it is only enabled by default in pycharm (and others). To enable it in vscode, you have to install _EditorConfig for VS Code_.

**Review by cristian-tamblay — APPROVED** [review]
Hope we can get some feedback from @Riul1999. Will approve to get things going

**Pull request merged**